colectare-date/colectare-date/Controllers/HartaController.cs

```diff
@@ -18,7 +21,148 @@ public IActionResult Harta()
                 .Where(c => c.TimpColectare.Date == dataCautata)
                 .ToList();
 
-            return View(colectari);
+            var rutaNeoptimizata = Enumerable.Range(0, colectari.Count + 2).ToList();
+            var matrice = IncarcaMatrice();
+            var rutaOptimizata = CalculeazaRutaOptimizata(matrice);
+
+            var distantaTraseuNeoptimizat = CalculeazaDistanta(rutaNeoptimizata, matrice);
+            var distantaTraseuOptimizat = CalculeazaDistanta(rutaOptimizata, matrice);
+
+            var timpEstimativ = rutaOptimizata.Count * 1;
+
+            var rutaFaraStartStop = rutaOptimizata.Where(index => index > 0 && index <= colectari.Count).ToList();
+            var colectariOrdonate = rutaFaraStartStop
+                .Select(index => colectari[index - 1])
+                .ToList();
+
+            var model = new RutaOptimizataViewModel
```

Comment:
Foarte bine ca am facut o clasa RutaOptimizataViewModel, dar o folosim doar aici, in ruta RutaOptimizata folosim partial clasa in combinatie cu ViewBag